docs/algorithms/Sorting Algorithms/HeapSort.md

```diff
@@ -0,0 +1,112 @@
+---
+
+id: selection-sort-algo  
+sidebar_position: 6  
+title: Heap Sort  
+sidebar_label: Heap Sort  
+
+---
+
+### Definition:
+
+Heap sort is a **comparison-based** sorting algorithm that uses a binary heap data structure to efficiently find the largest or smallest element, depending on the heap type. It is an **in-place** and **non-stable** sorting algorithm that operates by first building a max heap and then extracting elements from it.
+
+### Characteristics:
+
+- **Binary Heap**:
+  - Heap sort is based on a complete binary tree structure called a heap. The two types of heaps are:
+    - **Max Heap**: The root is the largest element.
+    - **Min Heap**: The root is the smallest element.
+
+- **In-Place Sorting**:
+  - Heap sort sorts the array in place without requiring additional memory beyond the array itself, making it memory efficient.
+
+- **Non-Stable**:
+  - Heap sort is a non-stable sorting algorithm, meaning that the relative order of equal elements may not be preserved during the sorting process.
+
+- **No Recursive Calls**:
+  - Unlike recursive algorithms like merge sort, heap sort uses an iterative approach to build the heap and extract elements.
+
+### Time Complexity:
+
+- **Best Case: O(n log n)**  
+  Heap sort involves building the heap (O(n)) and extracting elements from it (O(n log n)), so even in the best case, it requires O(n log n) time.
```

Comment:
replace `O(n)` with `$O(n)$` and `O(n log n)` with `$O(n log n)$`

Reply:
ok